### PR TITLE
[CI:DOCS] Update debian/ubuntu build instructions

### PR DESCRIPTION
**Pull request by andrewgdotcom**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Modern versions of Debian set the default gpg keyserver to `keys.openpgp.org`, but this keyserver strips userIDs that do not contain a (verified) email address. The PPA signing key mentioned in the user documentation has no verified (or verifiable) userIDs, so `keys.openpgp.org` serves it with no userIDs at all, and the result is not usable by gpg.

To fix, use a keyserver that serves all userIDs.

#### How to verify it

Go to a fresh Debian machine, and incant:

`gpg --recv-keys 0x018BA5AD9DF57A4448F0E6CF8BECF1637AD8C79D`

This will fail with the following error:

`gpg: key 8BECF1637AD8C79D: no user ID`

Now try:

`gpg --keyserver keyserver.ubuntu.com --recv-keys 0x018BA5AD9DF57A4448F0E6CF8BECF1637AD8C79D`

This will succeed.

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by andrewgdotcom** [comment]
Not sure who to ping for this, @lsm5 @rhatdan ?

**Comment by lsm5** [comment]
@andrewgdotcom the projectatomic ppa is old and unmaintained. It should actually be removed. I'm guessing debian has those dependencies by default. Could you please try installing the packages directly without adding that PPA? If that works, we can update the PR to get rid of 3 GPG+PPA lines and also remove the `stretch-backports`

**Comment by andrewgdotcom** [comment]
@lsm5 the Ubuntu Jammy instructions almost work on Debian Bookworm, except that golang-1.18 is not available, only golang-1.19

**Comment by andrewgdotcom** [comment]
buildah appears to build correctly under golang-1.19.  Do we really need to set $PATH explicitly?

**Comment by lsm5** [comment]
@andrewgdotcom GOPATH should be old stuff as well. Can you please try without setting GOPATH and also using the default `golang` package instead of `-1.18` / `-1.19` suffix. I'd prefer to keep the docs as generic as possible.

**Comment by andrewgdotcom** [comment]
It works perfectly on Debian Bookworm when using generic names and without custom PATHs etc.

But it doesn't build at all on Ubuntu Jammy because Jammy only has golang-1.18 and the code requires 1.19:

```
make[1]: Entering directory '/root/buildah/src/github.com/containers/buildah/tests/tools'
go build -o build/ginkgo ./vendor/github.com/onsi/ginkgo/ginkgo
go build -o build/go-md2man ./vendor/github.com/cpuguy83/go-md2man/v2
go build -o build/golangci-lint ./vendor/github.com/golangci/golangci-lint/cmd/golangci-lint
# honnef.co/go/tools/unused
vendor/honnef.co/go/tools/unused/unused.go:419:14: obj.Origin undefined (type *types.Var has no field or method Origin)
vendor/honnef.co/go/tools/unused/unused.go:421:14: obj.Origin undefined (type *types.Func has no field or method Origin)
note: module requires Go 1.19
# honnef.co/go/tools/staticcheck
vendor/honnef.co/go/tools/staticcheck/lint.go:4025:29: righti.Method(i).Origin undefined (type *types.Func has no field or method Origin)
vendor/honnef.co/go/tools/staticcheck/lint.go:4030:36: sel.Obj().(*types.Func).Origin undefined (type *types.Func has no field or method Origin)
note: module requires Go 1.19
make[1]: *** [Makefile:31: build/golangci-lint] Error 2
make[1]: Leaving directory '/root/buildah/src/github.com/containers/buildah/tests/tools'
make: *** [Makefile:126: install.tools] Error 2
```

**Comment by rhatdan** [comment]
@andrewgdotcom you need to sign your commit.

git commit -a --amend -s
git push --force

File: install.md
Comment (lsm5) [inline install.md:278]: Does the golang install need to be separate? I guess you can combine the golang install with the other packages in the previous line. Apart from that, LGTM

**Comment by lsm5** [comment]
@andrewgdotcom do you want to include the Ubuntu changes in this PR itself?

File: install.md
Comment (andrewgdotcom) [reply install.md:278]: I suppose not, I just kept it like that because it was that way before. I'll update.

**Comment by andrewgdotcom** [comment]
@lsm5 Is it OK to require that Ubuntu must be 22.10 kinetic or later? If so, I'll update accordingly.

File: install.md
Comment (andrewgdotcom) [reply install.md:278]: yes, skopeo has changed name. I've already updated that locally - it will be included in the updated commit.

**Comment by andrewgdotcom** [comment]
@rhatdan but all my commits are signed... 
Screenshot 2023-06-30 at 10 30 52

**Comment by rhatdan** [comment]
Well not according to our DCO checker.
@edsantiago @cevich Is this a problem in out checker?

**Comment by lsm5** [comment]
@andrewgdotcom your commit is GPG signed, but not signed-off using the lowercase `-s`

**Comment by lsm5** [comment]
@andrewgdotcom try a `git commit --amend -Ss`

**Comment by lsm5** [comment]
> @lsm5 Is it OK to require that Ubuntu must be 22.10 kinetic or later? If so, I'll update accordingly.

I'd be cool with having a separate section for 22.10 onwards, and leaving the 22.04 one as-is assuming many people are still on 22.04 ?

**Comment by andrewgdotcom** [comment]
@lsm5 But the 22.04 instructions no longer work as-is, because the minimum golang version required by the code has increased since this document was written. We would need to add specific instructions for people using 22.04 or before telling them how to upgrade to golang 1.19 by other means.

**Comment by lsm5** [comment]
> @lsm5 But the 22.04 instructions no longer work as-is, because the minimum golang version required by the code has increased since this document was written. We would need to add specific instructions for people using 22.04 or before telling them how to upgrade to golang 1.19 by other means.

ah got it. In that case we can leave the change as-is. If Ubuntu 22.04 users need specific help, we can handle those issues later.

We should be good to go once you add the sign-off to your commit. While you're at it, a nit: it might look nice to have the package dependencies listed in alphabetical order. Currently go-md2man and golang are listed at the end. Thanks

**Comment by lsm5** [comment]
@andrewgdotcom also, please rebase your branch on the latest `main` branch.

**Comment by andrewgdotcom** [comment]
damn, made a mess of that rebase sorry

**Review by lsm5 — APPROVED** [review]
/lgtm
/hold
/approve

**Comment by lsm5** [comment]
Thanks @andrewgdotcom !

**Pull request merged**
